core/arns/src/main/java/software/amazon/awssdk/arns/Arn.java

```diff
@@ -146,40 +146,51 @@ public static Builder builder() {
      *
      * @param arn - A string containing an Arn.
      * @return {@link Arn} - A modeled Arn.
+     * @throws IllegalArgumentException if {@code arn} cannot be parsed
+     * @throws NullPointerException if {@code arn} is {@code null}
      */
     public static Arn fromString(String arn) {
+        Validate.paramNotBlank(arn, "arn");
+        try {
+            return parseString(arn);
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("Malformed ARN: " + arn, e);
```

Comment:
Re-wrapping the exception again may make exception handling harder since users now need to check the cause of the exception. Can we just add the additional error message in the original exception?

Reply:
Sure.  Kind of a bit of copy & paste. New rev. posted.